src/components/slider/slider.ts

```diff
@@ -0,0 +1,250 @@
+import {
+  Component,
+  ElementRef,
+  HostBinding,
+  Input,
+  ViewEncapsulation,
+  AfterContentInit,
+} from '@angular/core';
+
+@Component({
+  moduleId: module.id,
+  selector: 'md-slider',
+  host: {
+    'tabindex': '0',
+    '(click)': 'onClick($event)',
+    '(drag)': 'onDrag($event)',
+    '(dragstart)': 'onDragStart($event)',
+    '(dragend)': 'onDragEnd()',
+    '(window:resize)': 'onResize()',
+    '(blur)': 'onBlur()',
+  },
+  templateUrl: 'slider.html',
+  styleUrls: ['slider.css'],
+  encapsulation: ViewEncapsulation.None,
+})
+export class MdSlider implements AfterContentInit {
+  /** A renderer to handle updating the slider's thumb and fill track. */
+  private _renderer: SliderRenderer = null;
+
+  /** The dimensions of the slider. */
+  private _sliderDimensions: ClientRect = null;
+
+  /** Whether or not the slider is disabled. */
+  private _disabled: boolean = false;
+
+  /** The miniumum value that the slider can have. */
+  private _min: number = 0;
+
+  /** The maximum value that the slider can have. */
+  private _max: number = 100;
+
+  /** The percentage of the slider that coincides with the value. */
+  private _percent: number = 0;
+
+  /**
+   * Whether or not the thumb is currently being dragged.
+   * Used to determine if there should be a transition for the thumb and fill track.
+   * @internal
```

Comment:
You can remove all use of `@internal`. We discovered last week that it breaks when used with the offline compiler.